### ISSUE TITLE
RecordLocator.TryGetSection by FormKey /w recordtype + loc return

### ISSUE THREAD

**Issue by Noggog** [Mutagen-Modding/Mutagen#67]

**Comment by Deigue** [comment]
Would this issue **now** entail RecordLocator.TryGetFormKey function for the end-user to call ?
Any more details or specifics about the requirements of this issue, aside the loc return?
- required params? any optional params?
- closest examples to follow suite with?

**Comment by Noggog** [comment]
Yeah, overall the RecordLocator just needs a lot of improvements.  It's a janky class that was made very early on just to get some work done.  It would be good to evolve into a polished front facing tool, though.

So this would be more of an API adjustment to `FileLocations`.  Also would now be FormKey based thanks to #111 

```
FileLocations locs = RecordLocator.GetFileLocations(path, constants);
locs.TryGetSection(FormKey key, RecordType type, out RangeInt64 section)
```

This would just be a wrapper around TryGetSection that exists, while also checking the RecordType is the one requested.  If the record types are different, return false as "not found".  Basically a way to query for a FormKey of an expected type.
